### PR TITLE
#75-EB Add new components to PromptPage and refactor into 2 separate tabs

### PR DESCRIPTION
**Pull request by eburdekin**
JIRA Ticket link: [PC-75](https://makeitmvp.atlassian.net/browse/PC-75)

## Type of change

- [ ] Bug fix
- [x] New feature
- [x] Refactoring

## Description

- Changes made? Created new components for Edit Post tab, Review Post tab, and added a PromptPageTabs component that will track which tab is active and change style accordingly.
- Reason for changes? Updating user flow per design team's mid-fi wireframe updates
- Behavior before? User prompting and response previewing/editing was all done on one page.
- Behavior after? User enters text, hits generate button, and is taken to the preview post page where they can view/edit the response. 
- How did you test? Tested successful API call and API call with error thrown.

## Checklist

- [x] I have performed a self-review of my code.
- [x]  I have manually tested the new feature and found it to be working as expected.
- [x] All commits in this PR are related to the issue.
- [x] Base branch for this PR is set to `develop`.
- [x] JIRA ticket # and my initials are noted in this PR's name (format example: #57-EB Add Pull Request template)

**Reminder**: every PR needs at least 1 peer approval before merging to the `develop` branch. Once a main ticket has been completed, along with all subtasks, associated code can be merged to `main`.

[PC-75]: https://makeitmvp.atlassian.net/browse/PC-75?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/PromptPage/PromptPage.jsx [hunk 1]
```diff
@@ -1,108 +1,45 @@
...
-      </div>
-      <div className="promptpage__preview-container">
-        {previewContent()}
-        <Button className="post" onClick={handlePostButtonClick}>
-          Post
-        </Button>
-      </div>
-    </div>
+    <>
+      <PromptPageTabs activeTab={activeTab} setActiveTab={setActiveTab} />
```
Comment (IguanasEverywhere) [inline src/pages/PromptPage/PromptPage.jsx:40]: Do we need to pass setActiveTab here? Is it being used in PromptPageTabs?

Comment (eburdekin) [reply src/pages/PromptPage/PromptPage.jsx:40]: You're right we don't need it. I'll remove!

File: src/components/ReviewPost/ReviewPost.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      <PreviewBox
+        previewText={previewText}
+        className="promptpage__preview-box"
+        onClickEdit={handleToggleEditing}
+      />
+    );
+  };
+
+  // placeholder to handle LinkedIn posting
+  const handlePostButtonClick = () => {};
```
Comment (IguanasEverywhere) [inline src/components/ReviewPost/ReviewPost.jsx:34]: Maybe we'll also need to update the activePage state here, though I know this is just a placeholder for now.

**Review by IguanasEverywhere — APPROVED** [review]
Looks great! Thank you for separating these into different tabs, it makes sense. As we chatted about, I think we can merge this to develop, then I'll start adding in the styling and components from my PR into the edit tab you've created here.

File: src/components/ReviewPost/ReviewPost.jsx (hunk 2 above)
Comment (eburdekin) [reply src/components/ReviewPost/ReviewPost.jsx:34]: Ah yes, Mohan's functions might take care of that for us too - I need to take a look

**Comment by eburdekin** [comment]
Thanks @IguanasEverywhere - sounds great!

**Pull request merged**